flink-cdc-runtime/src/main/java/com/ververica/cdc/runtime/operators/transform/TransformFilterProcessor.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2023 Ververica Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ververica.cdc.runtime.operators.transform;
+
+import com.ververica.cdc.common.data.RecordData;
+import com.ververica.cdc.common.data.binary.BinaryRecordData;
+import com.ververica.cdc.common.schema.Column;
+import com.ververica.cdc.runtime.parser.JaninoCompiler;
+import com.ververica.cdc.runtime.parser.TransformParser;
+import com.ververica.cdc.runtime.typeutils.DataTypeConverter;
+import org.codehaus.janino.ExpressionEvaluator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.List;
+
+/** The processor of the transform filter. It processes the data change event of matched table. */
+public class TransformFilterProcessor {
+    private static final Logger LOG = LoggerFactory.getLogger(TransformFilterProcessor.class);
+
+    public static boolean process(
+            BinaryRecordData after, TableInfo tableInfo, TransformFilter transformFilter) {
+        TransformExpressionKey transformExpressionKey =
+                generateTransformExpressionKey(tableInfo, transformFilter);
+        ExpressionEvaluator expressionEvaluator =
+                TransformExpressionCompiler.compileExpression(transformExpressionKey);
+        try {
+            return (Boolean)
+                    expressionEvaluator.evaluate(generateParams(after, tableInfo, transformFilter));
```

Comment:
Could one `TransformFilterProcessor` instance  holds a `expressionEvaluator` which built in construction? and  thus we can offer a method like `private boolean process( BinaryRecordData after)`. In this way, we  can avoid compile expression per record.